backend/src/main/kotlin/org/darkSolace/muse/userModule/model/User.kt

```diff
@@ -0,0 +1,57 @@
+package org.darkSolace.muse.userModule.model
+
+import org.hibernate.Hibernate
+import org.hibernate.annotations.Cascade
+import org.hibernate.annotations.CascadeType
+import java.util.*
+import javax.persistence.*
+
+@Entity
+@Table(name = "museUser", uniqueConstraints = [UniqueConstraint(columnNames = ["username", "email"])])
+data class User(
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE)
+    var id: Long? = null,
```

Comment:
The question mark means it may be null, doesn't it? Why may the Id be null?

Reply:
If we create a new user which is at this moment not already persisted in the DB it doesn't have an ID, in those cases the `id` would be `null`. The ID is assigned once the user is written to the DB.